src/coreclr/jit/lsraarm64.cpp

```diff
@@ -25,6 +25,415 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #include "sideeffects.h"
 #include "lower.h"
 
+//------------------------------------------------------------------------
+// getNextConsecutiveRefPosition: Get the next subsequent refPosition.
+//
+// Arguments:
+//    refPosition   - The refposition for which we need to find next refposition
+//
+// Return Value:
+//    The next refPosition or nullptr if there is not one.
+//
+RefPosition* LinearScan::getNextConsecutiveRefPosition(RefPosition* refPosition)
+{
+    assert(compiler->info.compNeedsConsecutiveRegisters);
+    RefPosition* nextRefPosition;
+    assert(refPosition->needsConsecutive);
+    nextConsecutiveRefPositionMap->Lookup(refPosition, &nextRefPosition);
+    assert((nextRefPosition == nullptr) || nextRefPosition->needsConsecutive);
+    return nextRefPosition;
+}
+
+//------------------------------------------------------------------------
+// setNextConsecutiveRegisterAssignment: For subsequent refPositions, set the register
+//   requirement to be the consecutive register(s) of the register that is assigned to
+//   the firstRefPosition.
+//   If one of the subsequent RefPosition is RefTypeUpperVectorRestore, sets the
+//   registerAssignment to not include any of the consecutive registers that are being
+//   assigned to the RefTypeUse refpositions.
+//
+// Arguments:
+//    firstRefPosition  - First refPosition of the series of consecutive registers.
+//    firstRegAssigned  - Register assigned to the first refposition.
```

Comment:
Not really. We call this method from places where `firstRegAssigned` is the one that corresponds to the Interval of the variable and we will just use it rather than assigning new register to the `firstRefPosition`. In such case `firstRefPosition->registerAssignment` might not reflect the `firstRegAssigned`.